### ISSUE TITLE
"Cannot read logs for Batch job" warning when tasks fail in google-batch

### ISSUE THREAD

**Issue by Puumanamana** [nextflow-io/nextflow#3772]
## Bug report 

Whenever a task fails on google-batch, I get the same warning: `Cannot read logs for Batch job [job_id]`. I'm not sure what this is about, but it makes the whole log output from nextflow harder to read (see example below). 

### Expected behavior and actual behavior

Actual: Warning keeps "polluting" stdout 
Expected: Either no issue with reading logs, or keep the warning in the nextflow.log (and not in stdout)

### Steps to reproduce the problem

Anytime an error occurs on google-batch, for example:

```nextflow
// main.nf

nextflow.enable.dsl = 2

process FAIL {
    tag "$v"

    input:
    val v

    script:
    """
    exit 1
    """
}

workflow {
    FAIL(Channel.of(1, 2))
}
```

```
// nextflow.config
workDir = "gs://nf-tower-public/scratch"

google {
    project = [GCP PROJECT]
    region = "us-central1"
    batch {
        bootDiskSize = 50.GB
        serviceAccountEmail = [SERVICE ACCOUNT]
        spot = true
    }
}

process {
    errorStrategy = "retry"
    maxRetries = 2
    executor  = "google-batch"
    container = "quay.io/biocontainers/fastp:0.23.2--h5f740d0_3"
    disk      = "50.GB"
}

```

### Program output 

```
// .nextflow.log
Mar-17 15:56:01.047 [main] DEBUG nextflow.cli.Launcher - $> nextflow main.nf
Mar-17 15:56:01.113 [main] INFO  nextflow.cli.CmdRun - N E X T F L O W  ~  version 23.02.1-edge
Mar-17 15:56:01.137 [main] DEBUG nextflow.plugin.PluginsFacade - Setting up plugin manager > mode=prod; embedded=false; plugins-dir=/home/cedric/.nextflow/plugins; core-plugins: nf-amazon@1.15.0,nf-azure@0.15.1,nf-codecommit@0.1.3,nf-console@1.0
.5,nf-ga4gh@1.0.4,nf-google@1.7.0,nf-tower@1.5.10,nf-wave@0.7.2
Mar-17 15:56:01.149 [main] INFO  org.pf4j.DefaultPluginStatusProvider - Enabled plugins: []
Mar-17 15:56:01.150 [main] INFO  org.pf4j.DefaultPluginStatusProvider - Disabled plugins: []
Mar-17 15:56:01.154 [main] INFO  org.pf4j.DefaultPluginManager - PF4J version 3.4.1 in 'deployment' mode
Mar-17 15:56:01.164 [main] INFO  org.pf4j.AbstractPluginManager - No plugins
Mar-17 15:56:01.190 [main] DEBUG nextflow.config.ConfigBuilder - Found config local: /home/cedric/sandbox/troubleshoot/stdout-polluted/nextflow.config
Mar-17 15:56:01.191 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /home/cedric/sandbox/troubleshoot/stdout-polluted/nextflow.config
Mar-17 15:56:01.215 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `standard`
Mar-17 15:56:01.936 [main] DEBUG nextflow.cli.CmdRun - Applied DSL=2 from script declararion
Mar-17 15:56:01.953 [main] INFO  nextflow.cli.CmdRun - Launching `main.nf` [backstabbing_brahmagupta] DSL2 - revision: a1c2370080
Mar-17 15:56:01.954 [main] DEBUG nextflow.plugin.PluginsFacade - Plugins default=[nf-google@1.7.0]
Mar-17 15:56:01.955 [main] DEBUG nextflow.plugin.PluginsFacade - Plugins resolved requirement=[nf-google@1.7.0, nf-tower@1.5.10]
Mar-17 15:56:01.955 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-google version: 1.7.0
Mar-17 15:56:01.965 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-google@1.7.0' resolved
Mar-17 15:56:01.965 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-google@1.7.0'
Mar-17 15:56:01.998 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-google@1.7.0
Mar-17 15:56:01.999 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-tower version: 1.5.10
Mar-17 15:56:02.000 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-tower@1.5.10' resolved
Mar-17 15:56:02.001 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-tower@1.5.10'
Mar-17 15:56:02.008 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-tower@1.5.10
Mar-17 15:56:02.022 [main] DEBUG nextflow.secret.LocalSecretsProvider - Secrets store: /home/cedric/.nextflow/secrets/store.json
Mar-17 15:56:02.026 [main] DEBUG nextflow.secret.SecretsLoader - Discovered secrets providers: [nextflow.secret.LocalSecretsProvider@75d4a80f] - activable => nextflow.secret.LocalSecretsProvider@75d4a80f
Mar-17 15:56:02.097 [main] DEBUG nextflow.Session - Session UUID: 2b7fe171-894c-40da-9707-b35a4ab167c5
Mar-17 15:56:02.097 [main] DEBUG nextflow.Session - Run name: backstabbing_brahmagupta
Mar-17 15:56:02.097 [main] DEBUG nextflow.Session - Executor pool size: 8
Mar-17 15:56:02.328 [main] DEBUG nextflow.util.ThreadPoolBuilder - Creating thread pool 'FileTransfer' minSize=10; maxSize=24; workQueue=LinkedBlockingQueue[10000]; allowCoreThreadTimeout=false
Mar-17 15:56:02.361 [main] DEBUG nextflow.cli.CmdRun -                                                                                                                                                                                                 Version: 23.02.1-edge build 5845
  Created: 26-02-2023 18:43 UTC
  System: Linux 5.15.0-1022-gcp                                                                                                                                                                                                                        Runtime: Groovy 3.0.15 on OpenJDK 64-Bit Server VM 17.0.3-internal+0-adhoc..src
  Encoding: UTF-8 (UTF-8)
  Process: 221022@nf-tower-main [10.128.0.2]
  CPUs: 8 - Mem: 62.8 GB (48 GB) - Swap: 0 (0)
Mar-17 15:56:02.397 [main] DEBUG nextflow.file.FileHelper - Can't check if specified path is NFS (1): gs://nf-tower-public/scratch

Mar-17 15:56:02.397 [main] DEBUG nextflow.Session - Work-dir: gs://nf-tower-public/scratch [null]                                                                                                                                            [85/134]Mar-17 15:56:02.398 [main] DEBUG nextflow.Session - Script base path does not exist or is not a directory: /home/cedric/sandbox/troubleshoot/stdout-polluted/bin
Mar-17 15:56:02.427 [main] DEBUG nextflow.executor.ExecutorFactory - Extension executors providers=[GoogleLifeSciencesExecutor, GoogleBatchExecutor]
Mar-17 15:56:02.442 [main] DEBUG nextflow.Session - Observer factory: DefaultObserverFactory
Mar-17 15:56:02.445 [main] DEBUG nextflow.Session - Observer factory: TowerFactory
Mar-17 15:56:02.465 [main] DEBUG nextflow.cache.CacheFactory - Using Nextflow cache factory: nextflow.cache.DefaultCacheFactory
Mar-17 15:56:02.476 [main] DEBUG nextflow.util.CustomThreadPool - Creating default thread pool > poolSize: 9; maxThreads: 1000
Mar-17 15:56:02.564 [main] DEBUG nextflow.Session - Session start
Mar-17 15:56:02.782 [main] DEBUG nextflow.script.ScriptRunner - > Launching execution
Mar-17 15:56:02.881 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: google-batch
Mar-17 15:56:02.881 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'google-batch'
Mar-17 15:56:02.884 [main] DEBUG nextflow.executor.Executor - [warm up] executor > google-batch
Mar-17 15:56:02.891 [main] DEBUG n.processor.TaskPollingMonitor - Creating task monitor for executor 'google-batch' > capacity: 1000; pollInterval: 10s; dumpInterval: 5m
Mar-17 15:56:02.897 [main] DEBUG nextflow.cloud.google.GoogleOpts - Google auth via application DEFAULT
Mar-17 15:56:02.903 [main] DEBUG n.c.google.batch.GoogleBatchExecutor - [GOOGLE BATCH] Executor config=BatchConfig[googleOpts=GoogleOpts(projectId:rome-pipeline-engine, credsFile:null, location:null, enableRequesterPaysBuckets:false, credentials
:ComputeEngineCredentials{transportFactoryClassName=com.google.auth.oauth2.OAuth2Utils$DefaultHttpTransportFactory})
Mar-17 15:56:02.918 [main] DEBUG n.c.google.batch.client.BatchClient - [GOOGLE BATCH] Creating service client with config credentials
Mar-17 15:56:03.584 [main] DEBUG nextflow.Session - Workflow process names [dsl2]: FAIL
Mar-17 15:56:03.585 [main] DEBUG nextflow.Session - Igniting dataflow network (2)
Mar-17 15:56:03.585 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > FAIL
Mar-17 15:56:03.586 [main] DEBUG nextflow.script.ScriptRunner - > Awaiting termination
Mar-17 15:56:03.586 [main] DEBUG nextflow.Session - Session await
Mar-17 15:56:06.253 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] submitted > job=nf-10d2a405-1679068564002; uid=nf-10d2a405-167906-0524b777-87b9-41800; work-dir=gs://nf-tower-public/scratch/10/d2a40504b1430d260815f4
54f961a7
Mar-17 15:56:06.253 [Task submitter] INFO  nextflow.Session - [10/d2a405] Submitted process > FAIL (2)
Mar-17 15:56:06.999 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] submitted > job=nf-b125e45e-1679068564002; uid=nf-b125e45e-167906-e97e5226-1f07-455e0; work-dir=gs://nf-tower-public/scratch/b1/25e45e2d880dfd703e0909
e366a533
Mar-17 15:56:07.000 [Task submitter] INFO  nextflow.Session - [b1/25e45e] Submitted process > FAIL (1)
Mar-17 15:57:42.961 [Task monitor] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Terminated job=nf-10d2a405-1679068564002; state=FAILED
Mar-17 15:57:44.062 [Task monitor] DEBUG n.processor.TaskPollingMonitor - Task completed > TaskHandler[id: 2; name: FAIL (2); status: COMPLETED; exit: 1; error: -; workDir: gs://nf-tower-public/scratch/10/d2a40504b1430d260815f454f961a7]
Mar-17 15:57:44.067 [Task monitor] DEBUG nextflow.processor.TaskProcessor - Handling unexpected condition for
  task: name=FAIL (2); work-dir=gs://nf-tower-public/scratch/10/d2a40504b1430d260815f454f961a7
  error [nextflow.exception.ProcessFailedException]: Process `FAIL (2)` terminated with an error exit status (1)
Mar-17 15:57:44.075 [Task monitor] INFO  nextflow.processor.TaskProcessor - [10/d2a405] NOTE: Process `FAIL (2)` terminated with an error exit status (1) -- Execution is retried (1)
Mar-17 15:57:44.188 [Task monitor] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] Terminated job=nf-b125e45e-1679068564002; state=FAILED
Mar-17 15:57:44.302 [Task monitor] WARN  n.c.g.batch.logging.BatchLogging - Cannot read logs for Batch job 'nf-b125e45e-167906-e97e5226-1f07-455e0' - cause: Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@76ed75b5[Not c
ompleted, task = java.util.concurrent.Executors$RunnableAdapter@7e267c00[Wrapped task = TrustedListenableFutureTask@4f2cfcec[status=PENDING, info=[task=[running=[NOT STARTED YET], com.google.api.gax.rpc.AttemptCallable@26d49194]]]]] rejected fro
m java.util.concurrent.ScheduledThreadPoolExecutor@3be6d114[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 0]
java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@76ed75b5[Not completed, task = java.util.concurrent.Executors$RunnableAdapter@7e267c00[Wrapped task = TrustedListenableFut
ureTask@4f2cfcec[status=PENDING, info=[task=[running=[NOT STARTED YET], com.google.api.gax.rpc.AttemptCallable@26d49194]]]]] rejected from java.util.concurrent.ScheduledThreadPoolExecutor@3be6d114[Terminated, pool size = 0, active threads = 0, q
ueued tasks = 0, completed tasks = 0]
        at java.base/java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2065)
        at java.base/java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:833)
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor.delayedExecute(ScheduledThreadPoolExecutor.java:340)
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor.schedule(ScheduledThreadPoolExecutor.java:562)
        at com.google.common.util.concurrent.MoreExecutors$ScheduledListeningDecorator.schedule(MoreExecutors.java:663)
        at com.google.api.gax.retrying.ScheduledRetryingExecutor.submit(ScheduledRetryingExecutor.java:116)
        at com.google.api.gax.retrying.CallbackChainRetryingFuture$AttemptCompletionListener.handle(CallbackChainRetryingFuture.java:137)
        at com.google.api.gax.retrying.CallbackChainRetryingFuture$AttemptCompletionListener.run(CallbackChainRetryingFuture.java:117)
        at com.google.common.util.concurrent.DirectExecutor.execute(DirectExecutor.java:31)
        at com.google.common.util.concurrent.AbstractFuture.executeListener(AbstractFuture.java:1270)
        at com.google.common.util.concurrent.AbstractFuture.complete(AbstractFuture.java:1038)
        at com.google.common.util.concurrent.AbstractFuture.setException(AbstractFuture.java:808)
        at com.google.api.core.AbstractApiFuture$InternalSettableFuture.setException(AbstractApiFuture.java:92)
        at com.google.api.core.AbstractApiFuture.setException(AbstractApiFuture.java:74)
        at com.google.api.gax.grpc.GrpcExceptionCallable$ExceptionTransformingFuture.onFailure(GrpcExceptionCallable.java:97)
        at com.google.api.core.ApiFutures$1.onFailure(ApiFutures.java:67)
        at com.google.common.util.concurrent.Futures$CallbackListener.run(Futures.java:1132)
        at com.google.common.util.concurrent.DirectExecutor.execute(DirectExecutor.java:31)
        at com.google.common.util.concurrent.AbstractFuture.executeListener(AbstractFuture.java:1270)
        at com.google.common.util.concurrent.AbstractFuture.complete(AbstractFuture.java:1038)
        at com.google.common.util.concurrent.AbstractFuture.setException(AbstractFuture.java:808)
        at io.grpc.stub.ClientCalls$GrpcFuture.setException(ClientCalls.java:574)
        at io.grpc.stub.ClientCalls$UnaryStreamToFuture.onClose(ClientCalls.java:544)
        at io.grpc.PartialForwardingClientCallListener.onClose(PartialForwardingClientCallListener.java:39)
        at io.grpc.ForwardingClientCallListener.onClose(ForwardingClientCallListener.java:23)
        at io.grpc.ForwardingClientCallListener$SimpleForwardingClientCallListener.onClose(ForwardingClientCallListener.java:40)
        at com.google.api.gax.grpc.ChannelPool$ReleasingClientCall$1.onClose(ChannelPool.java:541)
        at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:576)
        at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:70)
        at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:757)
        at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:736)
        at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
        at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
Mar-17 15:57:44.304 [Task monitor] WARN  n.c.g.batch.logging.BatchLogging - Cannot read logs for Batch job 'nf-b125e45e-167906-e97e5226-1f07-455e0' - cause: Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@4f850a16[Not c
ompleted, task = java.util.concurrent.Executors$RunnableAdapter@1e16b7b[Wrapped task = TrustedListenableFutureTask@3ed3b71e[status=PENDING, info=[task=[running=[NOT STARTED YET], com.google.api.gax.rpc.AttemptCallable@50b4ab9b]]]]] rejected from
 java.util.concurrent.ScheduledThreadPoolExecutor@3be6d114[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 0]
java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@4f850a16[Not completed, task = java.util.concurrent.Executors$RunnableAdapter@1e16b7b[Wrapped task = TrustedListenableFutu
reTask@3ed3b71e[status=PENDING, info=[task=[running=[NOT STARTED YET], com.google.api.gax.rpc.AttemptCallable@50b4ab9b]]]]] rejected from java.util.concurrent.ScheduledThreadPoolExecutor@3be6d114[Terminated, pool size = 0, active threads = 0, qu
eued tasks = 0, completed tasks = 0]
        at java.base/java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2065)
        at java.base/java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:833)
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor.delayedExecute(ScheduledThreadPoolExecutor.java:340)
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor.schedule(ScheduledThreadPoolExecutor.java:562)
        at com.google.common.util.concurrent.MoreExecutors$ScheduledListeningDecorator.schedule(MoreExecutors.java:663)
        at com.google.api.gax.retrying.ScheduledRetryingExecutor.submit(ScheduledRetryingExecutor.java:116)
        at com.google.api.gax.retrying.CallbackChainRetryingFuture$AttemptCompletionListener.handle(CallbackChainRetryingFuture.java:137)
        at com.google.api.gax.retrying.CallbackChainRetryingFuture$AttemptCompletionListener.run(CallbackChainRetryingFuture.java:117)
        at com.google.common.util.concurrent.DirectExecutor.execute(DirectExecutor.java:31)
        at com.google.common.util.concurrent.AbstractFuture.executeListener(AbstractFuture.java:1270)
        at com.google.common.util.concurrent.AbstractFuture.complete(AbstractFuture.java:1038)
        at com.google.common.util.concurrent.AbstractFuture.setException(AbstractFuture.java:808)
        at com.google.api.core.AbstractApiFuture$InternalSettableFuture.setException(AbstractApiFuture.java:92)
        at com.google.api.core.AbstractApiFuture.setException(AbstractApiFuture.java:74)
        at com.google.api.gax.grpc.GrpcExceptionCallable$ExceptionTransformingFuture.onFailure(GrpcExceptionCallable.java:97)
        at com.google.api.core.ApiFutures$1.onFailure(ApiFutures.java:67)
        at com.google.common.util.concurrent.Futures$CallbackListener.run(Futures.java:1132)
        at com.google.common.util.concurrent.DirectExecutor.execute(DirectExecutor.java:31)
        at com.google.common.util.concurrent.AbstractFuture.executeListener(AbstractFuture.java:1270)
        at com.google.common.util.concurrent.AbstractFuture.complete(AbstractFuture.java:1038)
        at com.google.common.util.concurrent.AbstractFuture.setException(AbstractFuture.java:808)
        at io.grpc.stub.ClientCalls$GrpcFuture.setException(ClientCalls.java:574)
        at io.grpc.stub.ClientCalls$UnaryStreamToFuture.onClose(ClientCalls.java:544)
        at io.grpc.PartialForwardingClientCallListener.onClose(PartialForwardingClientCallListener.java:39)
        at io.grpc.ForwardingClientCallListener.onClose(ForwardingClientCallListener.java:23)
        at io.grpc.ForwardingClientCallListener$SimpleForwardingClientCallListener.onClose(ForwardingClientCallListener.java:40)
        at com.google.api.gax.grpc.ChannelPool$ReleasingClientCall$1.onClose(ChannelPool.java:541)
        at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:576)
        at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:70)
        at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:757)
        at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:736)
        at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
        at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
Mar-17 15:57:44.304 [Task monitor] DEBUG n.processor.TaskPollingMonitor - Task completed > TaskHandler[id: 1; name: FAIL (1); status: COMPLETED; exit: 1; error: -; workDir: gs://nf-tower-public/scratch/b1/25e45e2d880dfd703e0909e366a533]
Mar-17 15:57:44.305 [Task monitor] DEBUG nextflow.processor.TaskProcessor - Handling unexpected condition for
  task: name=FAIL (1); work-dir=gs://nf-tower-public/scratch/b1/25e45e2d880dfd703e0909e366a533
  error [nextflow.exception.ProcessFailedException]: Process `FAIL (1)` terminated with an error exit status (1)
Mar-17 15:57:44.306 [Task monitor] INFO  nextflow.processor.TaskProcessor - [b1/25e45e] NOTE: Process `FAIL (1)` terminated with an error exit status (1) -- Execution is retried (1)
Mar-17 15:57:44.862 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] submitted > job=nf-bf493385-1679068664135; uid=nf-bf493385-167906-f5f7d756-176a-401d0; work-dir=gs://nf-tower-public/scratch/bf/4933857bc35124ecfc5cb6
a1c6ef47
Mar-17 15:57:44.863 [Task submitter] INFO  nextflow.Session - [bf/493385] Re-submitted process > FAIL (2)
Mar-17 15:57:45.492 [Task submitter] DEBUG n.c.g.batch.GoogleBatchTaskHandler - [GOOGLE BATCH] submitted > job=nf-71627721-1679068664374; uid=nf-71627721-167906-489d1895-1ba4-4c880; work-dir=gs://nf-tower-public/scratch/71/627721e7ce45bac50fae3a
663c95f7
Mar-17 15:57:45.492 [Task submitter] INFO  nextflow.Session - [71/627721] Re-submitted process > FAIL (1)
Mar-17 15:58:57.335 [SIGINT handler] DEBUG nextflow.Session - Session aborted -- Cause: SIGINT
Mar-17 15:58:57.351 [main] DEBUG nextflow.Session - Session await > all processes finished
Mar-17 15:58:57.351 [main] DEBUG nextflow.Session - Session await > all barriers passed
Mar-17 15:58:57.357 [main] WARN  n.processor.TaskPollingMonitor - Killing running tasks (2)
Mar-17 15:58:57.376 [main] DEBUG nextflow.trace.WorkflowStatsObserver - Workflow completed > WorkflowStats[succeededCount=0; failedCount=2; ignoredCount=0; cachedCount=0; pendingCount=0; submittedCount=2; runningCount=-2; retriesCount=2; aborted
Count=2; succeedDuration=0ms; failedDuration=1.2s; cachedDuration=0ms;loadCpus=-2; loadMemory=0; peakRunning=1; peakCpus=1; peakMemory=0; ]
Mar-17 15:58:57.385 [main] DEBUG nextflow.cache.CacheDB - Closing CacheDB done
Mar-17 15:58:57.385 [main] INFO  org.pf4j.AbstractPluginManager - Stop plugin 'nf-tower@1.5.10'
Mar-17 15:58:57.385 [main] DEBUG nextflow.plugin.BasePlugin - Plugin stopped nf-tower
Mar-17 15:58:57.385 [main] INFO  org.pf4j.AbstractPluginManager - Stop plugin 'nf-google@1.7.0'
Mar-17 15:58:57.385 [main] DEBUG nextflow.plugin.BasePlugin - Plugin stopped nf-google
Mar-17 15:58:57.400 [main] DEBUG nextflow.script.ScriptRunner - > Execution complete -- Goodbye
```

```
//stdout 

$ nextflow main.nf
N E X T F L O W  ~  version 23.02.1-edge
Launching `main.nf` [backstabbing_brahmagupta] DSL2 - revision: a1c2370080
executor >  google-batch (2)
[10/d2a405] process > FAIL (2) [ 33%] 1 of 3, failed: 1, retries: 1
[10/d2a405] NOTE: Process `FAIL (2)` terminated with an error exit status (1) -- Execution is retried (1)
WARN: Cannot read logs for Batch job 'nf-b125e45e-167906-e97e5226-1f07-455e0' - cause: Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@76ed75b5[Not completed, task = java.util.concurrent.Executors$RunnableAdapter@7e267cexecutor >  google-batch (2)
[b1/25e45e] process > FAIL (1) [ 50%] 2 of 4, failed: 2, retries: 2
[10/d2a405] NOTE: Process `FAIL (2)` terminated with an error exit status (1) -- Execution is retried (1)
WARN: Cannot read logs for Batch job 'nf-b125e45e-167906-e97e5226-1f07-455e0' - cause: Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@76ed75b5[Not completed, task = java.util.concurrent.Executors$RunnableAdapter@7e267cexecutor >  google-batch (3)
[bf/493385] process > FAIL (2) [ 50%] 2 of 4, failed: 2, retries: 2
[10/d2a405] NOTE: Process `FAIL (2)` terminated with an error exit status (1) -- Execution is retried (1)
WARN: Cannot read logs for Batch job 'nf-b125e45e-167906-e97e5226-1f07-455e0' - cause: Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@76ed75b5[Not completed, task = java.util.concurrent.Executors$RunnableAdapter@7e267cexecutor >  google-batch (4)
[71/627721] process > FAIL (1) [ 50%] 2 of 4, failed: 2, retries: 2
[10/d2a405] NOTE: Process `FAIL (2)` terminated with an error exit status (1) -- Execution is retried (1)
WARN: Cannot read logs for Batch job 'nf-b125e45e-167906-e97e5226-1f07-455e0' - cause: Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@76ed75b5[Not completed, task = java.util.concurrent.Executors$RunnableAdapter@7e267c00[Wrapped task = TrustedListenableFutureTask@4f2cfcec[status=PENDING, info=[task=[running=[NOT STARTED YET], com.google.api.gax.rpc.AttemptCallable@26d49194]]]]] rejected from java.util.concurrent.ScheduledThreadPoolExecutor@3be6d114[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 0]
WARN: Cannot read logs for Batch job 'nf-b125e45e-167906-e97e5226-1f07-455e0' - cause: Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@4f850a16[Not completed, task = java.util.concurrent.Executors$RunnableAdapter@1e16b7b[Wrapped task = TrustedListenableFutureTask@3ed3b71e[status=PENDING, info=[task=[running=[NOT STARTED YET], com.google.api.gax.rpc.AttemptCallable@50b4ab9b]]]]] rejected from java.util.concurrent.ScheduledThreadPoolExecutor@3be6d114[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 0]
[b1/25e45e] NOTE: Process `FAIL (1)` terminated with an error exit status (1) -- Execution is retried (1)
```

### Environment 

* Nextflow version: 23.02.1-edge build 5845
* Java version: 
```bash
$ java -version
openjdk version "17.0.3-internal" 2022-04-19
OpenJDK Runtime Environment (build 17.0.3-internal+0-adhoc..src)
OpenJDK 64-Bit Server VM (build 17.0.3-internal+0-adhoc..src, mixed mode, sharing)
```
* Operating system: [macOS, Linux, etc]
```bash
$ uname -a
Linux nf-tower-main 5.15.0-1022-gcp #29~20.04.1-Ubuntu SMP Sat Oct 29 18:17:56 UTC 2022 x86_64 GNU/Linux
```
* Bash version: (use the command `$SHELL --version`): zsh 5.8 (x86_64-ubuntu-linux-gnu)

**Comment by pditommaso** [comment]
Please include the complete log file

**Comment by Puumanamana** [comment]
I updated the post. Note that I interrupted the run (cf: SIGINT) at the end after the issue occurred

**Comment by pditommaso** [comment]
I'm not able to replicate the problem, but hopefully is going to be fixed by this change d7e38e9e that's going to be included in the upcoming edge release.

**Comment by Puumanamana** [comment]
It did work! Thanks for the fix.
